docs/runbooks/consumer-message-process-failures.md

```diff
@@ -0,0 +1,114 @@
+
+### Configuring Zed CLI
+
+To install the `zed` CLI, refer to [Authzed's README](https://github.com/authzed/zed?tab=readme-ov-file#getting-started) for specific instructions based on your OS
+
+In order to access SpiceDB using the `zed` cli, you'll need to configure a context for the SpiceDB endpoint. This will require a preshared token available in a Kubernetes secret on cluster.
+
+To configure the context:
+
+```shell
+# capture the preshared key from spicedb-config secret
+PSTK=$(oc get secret spicedb-config -o jsonpath='{.data.preshared_key}' | base64 -d)
+
+# create the context with zed cli -- change ENV to the ENV you are targeting (stage, prod)
+zed context set <ENV>-spicedb localhost:50051 $PSTK --insecure
+
+# verify the context was created and is the current context
+zed context list
+```
+
+To confirm the context is working:
+
+```shell
+# port-forwad the spicedb service to your system
+oc port-forward svc/kessel-relations-spicedb 50051:50051
+
+# check the schema
+zed schema read
+```
+
+The read command should dump out the entire SpiceDB schema. An error is provided if there are any connection issues or if no schema exists.
+
+
+## Schema Related Issues
+
+There are two known distinct reasons tuple creation could fail due to schema issues:
+1) The schema definitions between Inventory API and Relations API are not in sync
+2) The schema is valied but the tuple request values are malformed by Inventory API
+
+### Inventory Consumer fails to Create/Modify a Relationship due to schema mismatch
+
+**Example Error in Inventory API Logs**
+```
+msg=request failed: error creating tuple: rpc error: code = FailedPrecondition desc = error creating tuples: error writing relationships to SpiceDB: rpc error: code = FailedPrecondition desc = object definition `notifications/integration` not found
+```
+
+**Reason**
+The error `object definition "OBJECT" not found` indicates the schema loaded by SpiceDB does not contain a defintion for the tuple request. This is likely due to a mismatch between Inventory API and Relations API schema definitions
+
+**Verify**
+You can verify if there is a schema mismatch one of two ways:
+
+1) Check the loaded schema in Relations API for the object
+
+```shell
+zed schema read | grep <OBJECT>
+
+# using the example error above
+zed schema read | grep "notifications/integration"
+```
+
+If nothing is returned, the loaded schema is missing the expected definition
+
+2) Compare the loaded schema to the expected schema
+
+The schema is deployed via App Interface and is pulled from Github using a specifc commit hash (See [App Interface resource definition](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/data/services/insights/kessel/namespaces/kessel-prod.yml?ref_type=heads#L47)). When comparing the loaded schema to the expected schema, the commit ref listed in App Interface should be used
```

Comment:
We might want to avoid linking directly to a line here since there is a good chance this will change at some point. Maybe we could call out the specific configmap instead and just link to the file.

Reply:
Updated to call out the name of the configmap and linked to the actual resource file instead